File: src/Application Specific/Forms And Controls/Invoice/FRMChangeSageDate.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace FSM
{
    public partial class FRMChangeSageDate : FRMBaseForm, IForm
    {
        public FRMChangeSageDate()
        {
            InitializeComponent();
        }

        /* TODO ERROR: Skipped RegionDirectiveTrivia */
        private void LoadMe(object sender, EventArgs e)
        {
            dtpTaxDate.Value = Entity.Sys.Helper.MakeDateTimeValid(App.DB.ExecuteScalar("Select Name from tblpicklists where enumtypeid = 69", false, true));
            LoadForm(sender, e, this);
            dtpTaxDate.Enabled = false;
            btnSave.Enabled = false;
        }

        public IUserControl LoadedControl
        {
            get
            {
                return null;
            }
        }

        public void ResetMe(int newID)
        {
            ID = newID;
        }

        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        /* TODO ERROR: Skipped RegionDirectiveTrivia */
        private int _ID = 0;

        public int ID
        {
            get
            {
                return _ID;
            }

            set
            {
                _ID = value;
            }
        }

        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        /* TODO ERROR: Skipped RegionDirectiveTrivia */
        private void FRMChangeInvoicedTotal_Load(object sender, EventArgs e)
        {
            LoadMe(sender, e);
        }

        private void btnSave_Click(object sender, EventArgs e)
        {
            App.DB.ExecuteScalar("UPDATE tblpicklists SET Name = '" + Strings.Format(dtpTaxDate.Value, "yyyy-MM-dd") + "' where enumtypeid = 69", false, true);
            DialogResult = DialogResult.OK;
        }

        private void txtPassword_TextChanged(object sender, EventArgs e)
        {
            if ((Entity.Sys.Helper.HashPassword(txtPassword.Text.Trim()) ?? "") == (App.DB.Manager.Get().OverridePassword_Service ?? ""))
            {
                dtpTaxDate.Enabled = true;
                btnSave.Enabled = true;
            }
            else
            {
                dtpTaxDate.Enabled = false;
                btnSave.Enabled = false;
            }
        }

        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
    }
}

File: src/Application Specific/Forms And Controls/Invoice/FRMChangeSageDate.vb
Public Class FRMChangeSageDate : Inherits FSM.FRMBaseForm : Implements IForm

#Region "Interface Methods"

    Private Sub LoadMe(ByVal sender As Object, ByVal e As System.EventArgs) Implements IForm.LoadMe

        dtpTaxDate.Value = Entity.Sys.Helper.MakeDateTimeValid(DB.ExecuteScalar("Select Name from tblpicklists where enumtypeid = 69", False, True))

        LoadForm(sender, e, Me)

        Me.dtpTaxDate.Enabled = False
        Me.btnSave.Enabled = False

    End Sub

    Public ReadOnly Property LoadedControl() As IUserControl Implements IForm.LoadedControl
        Get
            Return Nothing
        End Get
    End Property

    Public Sub ResetMe(ByVal newID As Integer) Implements IForm.ResetMe
        ID = newID
    End Sub

#End Region

#Region "Properties"

    Private _ID As Integer = 0

    Public Property ID() As Integer
        Get
            Return _ID
        End Get
        Set(ByVal Value As Integer)
            _ID = Value

        End Set
    End Property

#End Region

#Region "Events"

    Private Sub FRMChangeInvoicedTotal_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        LoadMe(sender, e)
    End Sub

    Private Sub btnSave_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSave.Click

        DB.ExecuteScalar("UPDATE tblpicklists SET Name = '" & Format(dtpTaxDate.Value, "yyyy-MM-dd") & "' where enumtypeid = 69", False, True)

        Me.DialogResult = DialogResult.OK

    End Sub

    Private Sub txtPassword_TextChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles txtPassword.TextChanged
        If Entity.Sys.Helper.HashPassword(Me.txtPassword.Text.Trim) = DB.Manager.Get.OverridePassword_Service Then
            Me.dtpTaxDate.Enabled = True
            Me.btnSave.Enabled = True
        Else
            Me.dtpTaxDate.Enabled = False
            Me.btnSave.Enabled = False
        End If
    End Sub

#End Region

End Class